### ISSUE TITLE
Add a .gitignore file

### ISSUE THREAD

**Issue by necampanini** [jebert142/code-learning#2]
Currently, if you build the project, Visual Studio creates all the necessary files to build and run the project.

Git, by default, doesn't know any better and says 'hey, there are new untracked files here, do you want me to track them?'

No, we do not. Such is the case with a lot of the extraneous files that IDE's create as part of their build/run process. 

Visit www.gitignore.io and search for Visual Studio. You can take the result and create a new text file inside your current repository named '.gitignore'. 

The dot prefix is important. If Git sees there is a .gitignore file, it will go off its contents and essentially ignore those kinds of files found inside. This hides files from Git that other developers do not need to know about, keeping the focus on the base source code.

**Comment by jebert142** [comment]
I was tinkering with Visual Studio and I think I actually had it create a
local repository, which is why I don't think the commit I just added was
not shown on GitHub. I think I just committed to my local repository.

[image: Inline image 1]

But, the local repository creation added the .gitignore file that you just
referenced. I think that I messed up the repository connection though, or
maybe I just lost the connection to GitHub. How would I go about
reconnecting to the repository on GitHub so that I can push a new commit?

-Jon

**Comment by necampanini** [comment]
So I check documentation and stack overflow all the time for things like this, and all the things I can't recall off the top of my head, but know keywords around.

In this case, we want to make ensure the repository you pushed has maintained that connection to github.

'git remote show origin' will provide information on remote URLs and branch tracking. 

For me locally, my fetch and push URL are the same (by default they are and should usually stay the same), and point to 'https://github.com/jebert132/code-learning.git'

If this isn't the case for you, we need to see what has changed and from there we can re-add the remote URL.

**Comment by jebert142** [comment]
Just taking a couple steps back, if I need to upload a project to a repository, I should keep the project where it is located on my drive? Whether it be under Visual Studio > Projects or under my Projects folder on my C: drive. When I added the project to the repository, I dragged the files into the browser for upload, not by pushing the project into the repository. Is this why I'm not connected?

**Comment by necampanini** [comment]
Good catch, always take a step back to assure all base assumptions are verified.

In order to upload anything and communicate via repositories and their upstream connections, the project (or any files you want to share) has to be in the same folder/root structure as the .git folder.

I did not know github gave that upload functionality, but it only took the files you gave it, foregoing any actual git repository connecting.

Let's not do this now, but if you were starting from scratch this would be a good guide to follow:
https://help.github.com/articles/adding-an-existing-project-to-github-using-the-command-line/

But now that you have a repository up on GitHub, and a local repository, you can aim your local repository AT the existing one to enable a connection between the two.

For that, this documentation should help:
https://help.github.com/articles/adding-a-remote/

**Comment by jebert142** [comment]
Looks like the work I just did to connect the two didn't work. I'll try again and let you know what happens.
